src/Serializers/Orleans.Serialization.ProtobufNet/ProtobufNetSerializer.cs

```diff
@@ -69,6 +69,15 @@ public void Serialize(object item, ISerializationContext context, Type expectedT
                 throw new ArgumentNullException(nameof(context));
             }
 
+            // Length is not needed but some old serialized data still may
+            // have the length stamped, so we need to read and write it still.
+            if (item == null)
+            {
+                context.StreamWriter.Write(0);
+                return;
+            }
+
+            context.StreamWriter.Write(-1); // -1 should trigger an error in old deserializer
```

Comment:
... maybe I'll just use a `MemoryStream` and do `TryGetBuffer()` to get the supporting `ArraySegment<byte>` and then pass that over to the `IBinaryTokenStreamWriter`.. that way we should get minimal buffer copies while still being able to accurately write the length prefix

Reply:
.. and replace the `WriteAdapter`